src/node_jsvmapi.cc

```diff
@@ -514,6 +514,10 @@ namespace v8impl {
 #define CHECK_NEW_FROM_UTF8(isolate, result, str)                       \
     CHECK_NEW_FROM_UTF8_LEN((isolate), (result), (str), -1)
 
+#define GET_RETURN_STATUS()                                             \
+  v8impl::TryCatch::lastException().IsEmpty()                           \
```

Comment:
This won't work, because the static `TryCatch::lastException()` doesn't get set until the `TryCatch` instance on the stack gets destructed, which happens _after_ the return statement. Instead, check `tryCatch.HasCaught()`.